### ISSUE TITLE
Dynamically update a marker_layer?

### ISSUE THREAD

**Issue by anderl80** [pbugnion/gmaps#252]
The gmaps documentation shows the[ ACLED heatmap layer](http://jupyter-gmaps.readthedocs.io/en/latest/app_tutorial.html). This heatmap layer is dynamically updated when the attribute `locations` is fed with new information.

I tried to do the same with a marker_layer. I update the attribute `makers` with a list of new Markers objects `markers = [gmaps.Marker(location=tuple(*_last_pos), label=str(vehicle)]`.

However this is only rendered if I reload the (Jupyter Notebook) page (F5). The rest of the class structure is basically identical to the structure used in the ACLED example.

**Comment by pbugnion** [comment]
At the moment, dynamically updating markers is not supported. I agree that it would be a really good addition. I expect I'll get round to implementing this eventually, but till then, PRs welcome!

For anyone wanting to implement this, I think we should:
- move logic currently in `_marker_layer_options` into traitlets in the `Markers` widget. See for instance PR #211 that did the same for heatmaps. We need to preserve the current functionality, so the style traitlets (`hover_text`, `label`, `info_box_content` and `display_info_box`) will have to accept either a single argument that is broadcast to all markers, or a list, with one element per marker. We will probably have to use `validate` extensively at the `Markers` level.
- react to changes in the markers list in the frontend. See e.g. [the implementation for heatmaps](https://github.com/pbugnion/gmaps/blob/master/js/src/Heatmap.js#L81) for how to do this.

**Comment by pbugnion** [comment]
Fixed in PR #285. Using the following now updates the markers:

```py
marker_layer.markers = [gmaps.Marker(location=tuple(*_last_pos))]
```

See an [example application](https://jupyter-gmaps.readthedocs.io/en/latest/app_tutorial.html#updating-symbols-in-response-to-other-widgets) in the documentation.
